### PR TITLE
Seperated the network and container gateways

### PR DESCRIPTION
**Pull request by ktmitton**
Split out the network and container gateways into separate gateways.

### CODE REVIEW

File: Mittons.Fixtures/Docker/Gateways/IContainerGateway.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace Mittons.Fixtures.Docker.Gateways
+{
+    public interface IContainerGateway
+    {
+        Task<string> RunAsync(string imageName, string command, IEnumerable<KeyValuePair<string, string>> options, CancellationToken cancellationToken);
+
+        Task RemoveAsync(string containerId, CancellationToken cancellationToken);
+
+        Task<IPAddress> GetDefaultNetworkIpAddressAsync(string containerId, CancellationToken cancellationToken);
```
Comment (Frueber) [inline Mittons.Fixtures/Docker/Gateways/IContainerGateway.cs:14]: I was curious what the significance is of `Default` in the name here. Is this something you'd consider in the same context, or is this naming still inline what's intended? Default vs. something else?

`GetDefaultNetworkIpAddressAsync` vs `GetNetworkIpAddressAsync`

**Review by Frueber — APPROVED** [review]
Just had a minor question on a method name (`GetDefaultNetworkIpAddressAsync`). All else looks good to go to me.

File: Mittons.Fixtures/Docker/Gateways/IContainerGateway.cs (hunk 1 above)
Comment (ktmitton) [reply Mittons.Fixtures/Docker/Gateways/IContainerGateway.cs:14]: In this context, "default" means the default docker network the container is added to when first created, so it has meaning compared to every other network a container might later be connected to.

This is important in environments like running in an ADO docker environment, where the host can _only_ talk to the container across the default network.

Does that make sense?

Comment (Frueber) [reply Mittons.Fixtures/Docker/Gateways/IContainerGateway.cs:14]: 👍 makes sense

**Pull request merged**